SnapWar/Components/Common/NavButton.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  NavButton.swift
+//  SnapWar
+//
+//  Created by Pia Leung on 09/04/2020.
+//  Copyright © 2020 stickerwar. All rights reserved.
+//
+
+import UIKit
+
+class NavButton: UIButton {
+
+    var controller: UIViewController?
```

Comment:
i don't think this is the correct approach. i saw the description and i think i know what you're trying to do, but logically this is incorrect. only a view controller should be holding references to buttons, but not the other way around. this could introduce circular reference, which causes memory to not be released properly and thus memory leaks.